### PR TITLE
Directory endpoint

### PR DESCRIPTION
**Pull request by Kowalski1024**
Added:
* Models for Folder, File and FileTag
* Serializers to represent a directory
* directory endpoint

### CODE REVIEW

File: filechest/settings.py [hunk 1]
```diff
@@ -31,100 +32,107 @@
...
 
 # Default primary key field type
-# https://docs.djangoproject.com/en/4.1/ref/settings/#default-auto-field
+# https://docs.djangoproject.com/en/4.2/ref/settings/#default-auto-field
 
-DEFAULT_AUTO_FIELD = "django.db.models.BigAutoField"
+DEFAULT_AUTO_FIELD = 'django.db.models.BigAutoField'
 
-MEDIA_ROOT = BASE_DIR.joinpath("media")
+MEDIA_ROOT = os.path.join(BASE_DIR, 'media')
```
Comment (bubuss2) [inline filechest/settings.py:137]: Use pathlib's Path.joinpath() instead of os.path

File: filechest_server/templates/home.html [hunk 2]
```diff
@@ -18,4 +18,4 @@ <h1 class="text-center">FileChest</h1>
 
 </div>
 </body>
-</html>
+</html>
```
Comment (bubuss2) [inline filechest_server/templates/home.html:21]: missing newline

File: filechest_server/signals.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+    path = Path(settings.MEDIA_ROOT).joinpath(instance.path, instance.name)
+
+    if path.exists() and path.is_dir() and not any(path.iterdir()):
+        path.rmdir()
+
+
+@receiver(post_delete, sender=FileModel)
+def auto_delete_file_on_delete(sender, instance, **kwargs):
+    """
```
Comment (bubuss2) [inline filechest_server/signals.py:25]: missing typing

File: filechest_server/serializers.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+    def get_filename(self, obj):
+        return obj.filename()
+
+    def get_tags(self, obj):
+        return [tag.name for tag in obj.tags.all()]
+
+    def get_size(self, obj):
+        return obj.file.size
+
```
Comment (bubuss2) [inline filechest_server/serializers.py:47]: missing typing

File: filechest_server/serializers.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    @staticmethod
+    def get_subfolders(obj):
+        return [folder.name for folder in obj.subfolders.all()]
+
+    @staticmethod
+    def get_files(obj):
+        files = obj.files.all()
+        serializer = FileSerializer(files, many=True)
+        return serializer.data
```
Comment (bubuss2) [inline filechest_server/serializers.py:21]: missing typing again

File: filechest/settings.py [hunk 6]
```diff
@@ -31,100 +32,107 @@
...
 # Static files (CSS, JavaScript, Images)
-# https://docs.djangoproject.com/en/4.1/howto/static-files/
+# https://docs.djangoproject.com/en/4.2/howto/static-files/
 
-STATIC_URL = "static/"
+STATIC_URL = 'static/'
 
 # Default primary key field type
-# https://docs.djangoproject.com/en/4.1/ref/settings/#default-auto-field
+# https://docs.djangoproject.com/en/4.2/ref/settings/#default-auto-field
```
Comment (bubuss2) [inline filechest/settings.py:134]: Do we want to be consistent with "" or switch to ''?

File: filechest/settings.py (hunk 1 above)
Comment (Kowalski1024) [reply filechest/settings.py]: done

File: filechest_server/serializers.py (hunk 4 above)
Comment (Kowalski1024) [reply filechest_server/serializers.py]: done

File: filechest_server/serializers.py (hunk 5 above)
Comment (Kowalski1024) [reply filechest_server/serializers.py]: done

File: filechest/settings.py (hunk 6 above)
Comment (Kowalski1024) [reply filechest/settings.py]: done

**Review by bubuss2 — APPROVED** [review]

**Pull request merged**